client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/pattern-list.tsx

```diff
@@ -1,57 +1,73 @@
-import { Button } from '@automattic/components';
-import { useLocale } from '@automattic/i18n-utils';
-import classnames from 'classnames';
-import { useSite } from '../../../../hooks/use-site';
-import PatternPreviewAutoHeight from './pattern-preview-auto-height';
-import { getPatternPreviewUrl } from './utils';
-import type { Pattern } from './types';
+import { isEnabled } from '@automattic/calypso-config';
+import AnimateWithFeatures from './animate-with-features';
+import useSectionPatternsMapByCategory from './hooks/use-sections-map-by-category';
+import PatternSelector from './pattern-selector';
+import type { Pattern, Category } from './types';
 import './pattern-list.scss';
 
-interface Props {
-	stylesheet: string;
+type PatternListProps = {
+	onSelect: ( selectedPattern: Pattern | null ) => void;
 	patterns: Pattern[];
-	shownPatterns: Pattern[];
 	selectedPattern: Pattern | null;
-	activeClassName: string;
-	onSelect: ( selectedPattern: Pattern | null ) => void;
-}
+	categories: Category[];
+	categorySelected: string | null;
+	openPatternList: boolean | null;
+	onDoneClick?: () => void;
+};
 
 const PatternList = ( {
-	stylesheet,
+	onSelect,
 	patterns,
-	shownPatterns,
 	selectedPattern,
-	activeClassName,
-	onSelect,
-}: Props ) => {
-	const locale = useLocale();
-	const site = useSite();
+	categories,
+	categorySelected,
+	openPatternList,
+	onDoneClick,
+}: PatternListProps ) => {
+	const sectionPatternsMapByCategory = useSectionPatternsMapByCategory( patterns, categories );
+	const categoryPatterns = categorySelected ? sectionPatternsMapByCategory[ categorySelected ] : [];
+
+	if ( ! isEnabled( 'pattern-assembler/categories' ) ) {
+		return (
+			<PatternSelector
+				showHeader
+				showDoneButton
+				patterns={ patterns }
+				selectedPattern={ selectedPattern }
+				onDoneClick={ onDoneClick }
+				onSelect={ onSelect }
+			/>
+		);
+	}
 
 	return (
-		<>
-			{ patterns?.map( ( pattern: Pattern, index: number ) => (
-				<PatternPreviewAutoHeight
-					key={ `${ index }-${ pattern.id }` }
-					url={ getPatternPreviewUrl( {
-						id: pattern.id,
-						language: locale,
-						siteTitle: site?.name,
-						stylesheet,
-					} ) }
-					patternId={ pattern.id }
-					patternName={ pattern.category }
-					isShown={ shownPatterns.includes( pattern ) }
-				>
-					<Button
-						title={ pattern.category }
-						className={ classnames( 'pattern-list__pattern', {
-							[ activeClassName ]: pattern.id === selectedPattern?.id,
-						} ) }
-						onClick={ () => onSelect( pattern ) }
-					/>
-				</PatternPreviewAutoHeight>
-			) ) }
-		</>
+		<AnimateWithFeatures featureName="domAnimation">
```

Comment:
If we do it this way, we will load the `framer-motion` when user lands on the pattern assembler, and that's why we did it via `AsyncLoad`. You can see the size changes a lot (maybe not just related to this one)

```
with-theme-assembler-flow     +899551 B  (+393.3%)  +267870 B  (+396.8%)
import-flow                   +868420 B   (+46.6%)  +257083 B   (+55.7%)
site-setup-flow               +832975 B   (+42.2%)  +244793 B   (+49.5%)
```